core/common/src/main/java/alluxio/underfs/BaseUnderFileSystem.java

```diff
@@ -75,6 +75,11 @@ public boolean isObjectStorage() {
     return false;
   }
 
+  @Override
+  public boolean isSeekable() {
+    return false;
```

Comment:
This is probably `true`, but not related to this optimization

Reply:
The base fs is the abstract fs with default impls. So I think it should be false, but I think the `localFileSystem` can be true.

Reply:
Ah okay, you're right. For some reason I thought this was `LocalUnderFileSystem`.

Reply:
`LocalFileSysytem` currently returns `FileInputStream`, so it's does not support seek, unless we return `RandomAccessFile`.

Reply:
We could write a wrapper around it which implements `seek` using `skip`, since I think `skip` is implemented through a `seek` call in a `FileInputStream`. We would need to verify this is optimal, but this is can be done in a separate PR.

Reply:
But that only works to skip forward not backward.

Reply:
Yeah for seek backward we would need to use `getChannel` and operate on that.